Build failure: postgresql12Packages.repmgr

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `postgresql12Packages.repmgr`

### Build log

```
repmgr> unpacking sources
repmgr> unpacking source archive /nix/store/xy6cbh2jwqxlz8xd4fcgmp7n5g201hg7-source
repmgr> source root is source
repmgr> patching sources
repmgr> updateAutotoolsGnuConfigScriptsPhase
repmgr> configuring
repmgr> configure flags: --prefix=/nix/store/0ysfziwq6513v6gbrkzgk0nnz3jra7zp-repmgr-5.4.1
repmgr> checking for a sed that does not truncate output... /nix/store/zl7fpambb4db6by8mhl9xcmr6n1pp98c-gnused-4.9/bin/sed
repmgr> checking for pg_config... /nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/bin/pg_config
repmgr> configure: building against PostgreSQL 12.15
repmgr> checking for gnused... no
repmgr> checking for gsed... no
repmgr> checking for sed... yes
repmgr> configure: creating ./config.status
repmgr> config.status: creating Makefile
repmgr> config.status: creating Makefile.global
repmgr> config.status: creating config.h
repmgr> building
repmgr> build flags: SHELL=/nix/store/ir0j7zqlw9dc49grmwplppc7gh0s40yf-bash-5.2-p15/bin/bash
repmgr> Building against PostgreSQL 12
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr.o repmgr.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -shared -o repmgr.so repmgr.o -lcurl -ljson-c -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib  -L/nix/store/s9fs1xciw3m79d9qhg0zdglb4qji5i8l-libxml2-2.11.4/lib  -Wl,--as-needed -Wl,-rpath,'/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib',--enable-new-dtags  -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib -lpq
repmgr> sed -E 's/REPMGR_VERSION_DATE.*""/REPMGR_VERSION_DATE "2023-08-13"/' repmgr_version.h.in >repmgr_version.h; \
repmgr> sed -i -E 's/PG_ACTUAL_VERSION_NUM/PG_ACTUAL_VERSION_NUM 120015/' repmgr_version.h
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-client.o repmgr-client.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-action-primary.o repmgr-action-primary.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-action-standby.o repmgr-action-standby.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-action-witness.o repmgr-action-witness.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-action-cluster.o repmgr-action-cluster.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-action-node.o repmgr-action-node.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-action-service.o repmgr-action-service.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o repmgr-action-daemon.o repmgr-action-daemon.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o configdata.o configdata.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o configfile.o configfile.c
repmgr> flex  -o'configfile-scan.c' configfile-scan.l
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o configfile-scan.o configfile-scan.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o log.o log.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o strutil.o strutil.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o controldata.o controldata.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o dirutil.o dirutil.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o compat.o compat.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o dbutils.o dbutils.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o sysutils.o sysutils.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o pgbackupapi.o pgbackupapi.c
repmgr> cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC repmgr-client.o repmgr-action-primary.o repmgr-action-standby.o repmgr-action-witness.o repmgr-action-cluster.o repmgr-action-node.o repmgr-action-service.o repmgr-action-daemon.o configdata.o configfile.o configfile-scan.o log.o strutil.o controldata.o dirutil.o compat.o dbutils.o sysutils.o pgbackupapi.o -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib -lpgcommon -lpgport -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib -lpq -lcurl -ljson-c -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib  -L/nix/store/s9fs1xciw3m79d9qhg0zdglb4qji5i8l-libxml2-2.11.4/lib  -Wl,--as-needed -Wl,-rpath,'/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib',--enable-new-dtags  -lpgcommon -lpgport -lxml2 -lssl -lcrypto -lgssapi_krb5 -lz -lreadline -lcrypt -lm  -o repmgr
repmgr> /nix/store/fm1xq4f3g48k1mmwl6qg9jc3ipgfsk1x-binutils-2.40/bin/ld: cannot find -lcrypt: No such file or directory
repmgr> collect2: error: ld returned 1 exit status
repmgr> make: *** [Makefile:84: repmgr] Error 1
error: builder for '/nix/store/zfm903dc7hhbc4bvd0h51qniqhmsypyl-repmgr-5.4.1.drv' failed with exit code 2;
       last 10 log lines:
       > cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o controldata.o controldata.c
       > cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o dirutil.o dirutil.c
       > cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o compat.o compat.c
       > cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o dbutils.o dbutils.c
       > cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o sysutils.o sysutils.c
       > cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC -std=gnu89 -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include -Wall -Wmissing-prototypes -Wmissing-declarations  -I. -I./ -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/server -I/nix/store/qii9n5fc2cr53qq437n2gmfb98sjiv36-postgresql-12.15/include/internal -I/nix/store/111xkqxyjv4dqap3fj2d745idccgg1fh-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/8xl8nwa8iizcclmxi4zdh5jipr4gijzv-libxml2-2.11.4-dev/include/libxml2   -c -o pgbackupapi.o pgbackupapi.c
       > cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2 -fPIC repmgr-client.o repmgr-action-primary.o repmgr-action-standby.o repmgr-action-witness.o repmgr-action-cluster.o repmgr-action-node.o repmgr-action-service.o repmgr-action-daemon.o configdata.o configfile.o configfile-scan.o log.o strutil.o controldata.o dirutil.o compat.o dbutils.o sysutils.o pgbackupapi.o -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib -lpgcommon -lpgport -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib -lpq -lcurl -ljson-c -L/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib  -L/nix/store/s9fs1xciw3m79d9qhg0zdglb4qji5i8l-libxml2-2.11.4/lib  -Wl,--as-needed -Wl,-rpath,'/nix/store/fpndwjxlhp200rv0zqgvvmqp464rybf7-postgresql-12.15-lib/lib',--enable-new-dtags  -lpgcommon -lpgport -lxml2 -lssl -lcrypto -lgssapi_krb5 -lz -lreadline -lcrypt -lm  -o repmgr
       > /nix/store/fm1xq4f3g48k1mmwl6qg9jc3ipgfsk1x-binutils-2.40/bin/ld: cannot find -lcrypt: No such file or directory
       > collect2: error: ld returned 1 exit status
       > make: *** [Makefile:84: repmgr] Error 1
       For full logs, run 'nix log /nix/store/zfm903dc7hhbc4bvd0h51qniqhmsypyl-repmgr-5.4.1.drv'.
```


### Notify maintainers
@zimbatm

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.39, NixOS, 23.11 (Tapir), 23.11pre506474.12303c652b8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/servers/sql/postgresql/ext/repmgr.nix
+++ b/pkgs/servers/sql/postgresql/ext/repmgr.nix
@@ -8,6 +8,7 @@
 , flex
 , curl
 , json_c
+, libxcrypt
 }:
 
 stdenv.mkDerivation rec {
@@ -23,7 +24,8 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ flex ];
 
-  buildInputs = [ postgresql openssl zlib readline curl json_c ];
+  buildInputs = [ postgresql openssl zlib readline curl json_c ]
+    ++ lib.optionals (stdenv.isLinux && lib.versionOlder postgresql.version "13") [ libxcrypt ];
 
   installPhase = ''
     mkdir -p $out/{bin,lib,share/postgresql/extension}

